modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -132,10 +133,27 @@
 
       (isa? coercion :Coercion/YYYYMMDDHHMMSSString->Temporal)
       {"$dateFromString" {:dateString field-name
-                          :format "%Y%m%d%H%M%S"
-                          :onError field-name}}
+                          :format     "%Y%m%d%H%M%S"
+                          :onError    field-name}}
 
-      :else field-name)))
+      ;; mongo only supports datetime
+      (isa? coercion :Coercion/ISO8601->DateTime)
+      {"$dateFromString" {:dateString field-name
+                          :onError    field-name}}
+
+
+      (isa? coercion :Coercion/ISO8601->Date)
+      (throw (ex-info (tru "Driver Mongo does not support {0}" coercion)
+                      {:type              qp.error-type/unsupported-feature
+                       :coercion-strategy coercion}))
```

Comment:
I'm confused. [`$dateFromString` can definitely parse stuff in `YYYY-MM-DD` format.(https://www.mongodb.com/docs/v5.0/reference/operator/aggregation/dateFromString/)

Reply:
yes but it requires the source to have `YYYY-MM-DD` format and the resulted data type is still a Datetype.

Maybe this is the question of what does cast `ISO8601->Date` mean? 
1. does it mean it takes a field with ISO8601 format (either DateTime or Date format) and uses it as a `Date` type?
2. Or it means taking a field with ISO8601 with Date format

I'm inclined to believe it is the 1st, because it makes cast consistency with SQL => We can have a text column like `2022-05-13T09:57:17+00:00`, and we could use it as a Date or Datetime by using cast.

Reply:
but I could see supporting the `YYYY-MM-DD` format would be useful though.

Reply:
FWIW, I consulted this with @luizarakaki and he agrees mongo shouldn't have support casting to `date` and `time`.

https://metaboat.slack.com/archives/C03SHMZ266R/p1662980879849149